src/main/java/com/nedap/archie/serializer/adl/ADLDefinitionSerializer.java

```diff
@@ -35,13 +38,26 @@ public ADLDefinitionSerializer(ADLStringBuilder builder) {
         constraintSerializers.put(CTime.class, new CTimeSerializer(this));
     }
 
+    public static String serialize(CObject cons) {
+        final ADLStringBuilder builder = new ADLStringBuilder();
+        ADLDefinitionSerializer serializer = new ADLDefinitionSerializer(builder);
+        serializer.appendCObject(cons);
+        return builder.toString();
+    }
 
     public ADLStringBuilder getBuilder() {
         return builder;
     }
 
     public String getTermText(CObject cobj) {
-        ArchetypeTerm term = cobj.getArchetype().getTerm(cobj, cobj.getArchetype().getOriginalLanguage().getCodeString());
+        String originalLanguage = ofNullable(cobj)
+                .flatMap(c -> ofNullable(c.getArchetype()))
+                .flatMap(a -> ofNullable(a.getOriginalLanguage()))
```

Comment:
You can just use .map instead of flatmap and custom wrapping the result in an Optional:

        String originalLanguage = ofNullable(cobj)
            .map(CObject::getArchetype)
            .map(Archetype::getOriginalLanguage)
            .map(TerminologyCode::getCodeString)
            .orElse(null);

Apart from that, it looks good 👍